paper-api/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java

```diff
@@ -143,49 +145,116 @@ public Player getEntity() {
 
     // Paper end - improve death events
 
-    // Paper start - adventure
     /**
-     * Set the death message that will appear to everyone on the server.
+     * Get whether the death message should be shown.
+     * By default, this is determined by {@link org.bukkit.GameRule#SHOW_DEATH_MESSAGES}.
+     *
+     * @return whether the death message should be shown
+     * @see #deathChatMessage()
+     * @see #deathScreenMessage()
+     */
+    public boolean shouldDisplayDeathMessage() {
+        return displayDeathMessage;
+    }
+
+    /**
+     * Set whether the death message should be shown.
+     * By default, this is determined by {@link org.bukkit.GameRule#SHOW_DEATH_MESSAGES}.
+     *
+     * @param displayDeathMessage whether the death message should be shown
+     * @see #deathChatMessage()
+     * @see #deathScreenMessage()
+     */
+    public void setShouldDisplayDeathMessage(boolean displayDeathMessage) {
+        this.displayDeathMessage = displayDeathMessage;
+    }
+
+    /**
+     * Set the death message that will appear in chat and on death screen to everyone on the server.
```

Comment:
Sounds a bit weird imo.

```suggestion
     * Set the death message that will be broadcasted and appear on the players death screen.
```